### PR TITLE
Support MongoDBStore

### LINKED ISSUE #4: Support MongoDB
Some users prefer using NoSQL, one of the most popular one is MongoDB. Is there any chance this one will be implemented?

### PR DESCRIPTION
**Pull request by JustMangoT**
This PR adds MongoDB as an option for storing sessions.

closes #4

### CODE REVIEW

**Comment by maxcountryman** [comment]
Wonderful! Thank you for this.

**Comment by maxcountryman** [comment]
It looks like we'll want to run cargo +nightly fmt to pass CI.

**Comment by maxcountryman** [comment]
We're getting closer--it looks like the integration tests might not be passing.

**Comment by JustMangoT** [comment]
Yeah, I'm investigating why it does not work, it's been 1hr but still no progress

**Comment by maxcountryman** [comment]
Does it work for you locally?

I'm not too familiar with Mongo unfortunately, but I wonder if there could be some difference between the container's version of Mongo and a local version.

**Comment by JustMangoT** [comment]
Unfortunately, it doesn't work locally. My guess is that Deserializing Object from MongoDB to Rust SessionRecord maybe the cause

**Comment by maxcountryman** [comment]
> My guess is that Deserializing Object from MongoDB to Rust SessionRecord maybe the cause

I'm not sure exactly how you're testing, but if you're using the `HandleErrorLayer`, you can bind `BoxError` to something like `err` and `dbg!(err)` to get a better idea of what's failing.

**Comment by JustMangoT** [comment]
Tests passed on local

File: src/mongodb_store.rs [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+#[async_trait]
+impl SessionStore for MongoDBStore {
+    type Error = MongoDBStoreError;
+
+    async fn save(&self, session_record: &SessionRecord) -> Result<(), Self::Error> {
+        self.col()
+            .update_one(
+                doc! {
+                    "_id": session_record.id().as_uuid()
```
Comment (maxcountryman) [inline src/mongodb_store.rs:77]: Being able to use the native type seems quite nice! This seems good to me.

Comment (JustMangoT) [reply src/mongodb_store.rs:77]: Also I noticed that sqlx providers don't use native uuid (binary), or maybe it needs to be stored as string? I'm not familiar with uuid in sql.

Comment (maxcountryman) [reply src/mongodb_store.rs:77]: At least with Postgres, there's a module you can enable to support UUID as a column type. But I'm not sure if it's available by default these days and so it's not something that's currently being used.

**Pull request merged**